include/boost/gil/image_processing/numeric.hpp

```diff
@@ -295,6 +295,120 @@ inline void compute_hessian_entries(
     detail::convolve_2d(dy, sobel_y, ddyy);
 }
 
+namespace detail {
+enum class direction: std::size_t {
```

Comment:
Why `std::size_t`? Why not `unsigned int` or `std::uint32_t`?

Reply:
It is an index into `std::vector`. But yes, any unsigned type will do. Though the main problem is verbosity, which I don't think I can get rid of.

Reply:
See my https://github.com/boostorg/gil/pull/500#discussion_r436177784